lib/api/controllers/adminController.js

```diff
@@ -114,27 +114,28 @@ function AdminController(kuzzle) {
       .then(newRequestObject => {
         var
           pluginsConfig = {},
-          response;
+          promises = [];
 
         modifiedRequestObject = newRequestObject;
 
         _.forEach(kuzzle.pluginsManager.plugins, (object, id) => {
-          pluginsConfig[id] = _.assignIn({}, object);
-          if(pluginsConfig[id].object.context && pluginsConfig[id].object.context.config) {
-            // remove the kuzzle configuration which can be included into plugin context
-            delete pluginsConfig[id].object.context.config;
-          }
+          promises.push(kuzzle.internalEngine.get('plugins', id));
```

Comment:
nitpicking: `_.forEach` is not needed anymore since you're only using the object key.

I would rewrite this using a mere `Object.keys` call.